.cursor/rules/responsive-design-patterns.mdc

```diff
@@ -0,0 +1,167 @@
+---
+alwaysApply: true
+---
+
+# Responsive Design: Container Queries Over Media Queries
+
+## Principle
+
+Use container-based styling and `useContainerBreakpoints` instead of media queries for responsive behavior. This creates more maintainable and predictable responsive components.
+
+## Guidelines
+
+### Do This ✅
+
+```typescript
+// Good: Use useContainerBreakpoints hook
+import { useContainerBreakpoints } from '@/hooks/useContainerBreakpoints'
+
+const PayrollReceipts = () => {
+  const { isSmall, isMedium, isLarge } = useContainerBreakpoints()
+
+  return (
+    <div className={clsx(
+      'payroll-receipts',
+      {
+        'payroll-receipts--compact': isSmall,
+        'payroll-receipts--expanded': isLarge
+      }
+    )}>
+      {/* Component content */}
+    </div>
+  )
+}
+
+// Good: CSS container queries
+.payroll-container {
+  container-type: inline-size;
+}
+
+@container (max-width: 768px) {
+  .payroll-receipts {
+    flex-direction: column;
+    gap: 8px;
+  }
+}
+
+@container (min-width: 769px) {
+  .payroll-receipts {
+    flex-direction: row;
+    gap: 16px;
+  }
+}
+```
```

Comment:
For this one we need to just provide instructions to use the `Grid` and `Flex` components